plugins/PostProcessingPlugin/scripts/AddCoolingProfile.py

```diff
@@ -0,0 +1,877 @@
+                            current_fan_speed = str(fan_sp_skirt.split("S")[1])
+                        elif ";TYPE:WALL-INNER" in line:
+                            modified_data += line + "\n"
+                            modified_data += fan_sp_wall_inner + this_fan + "\n"
+                            current_fan_speed = str(fan_sp_wall_inner.split("S")[1])
+                        elif ";TYPE:WALL-OUTER" in line:
+                            modified_data += line + "\n"
+                            modified_data += fan_sp_wall_outer + this_fan + "\n"
+                            current_fan_speed = str(fan_sp_wall_outer.split("S")[1])
+                        elif ";TYPE:FILL" in line:
+                            modified_data += line + "\n"
+                            modified_data += fan_sp_fill + this_fan + "\n"
+                            current_fan_speed = str(fan_sp_fill.split("S")[1])
+                        elif ";TYPE:SKIN" in line:
+                            modified_data += line + "\n"
+                            modified_data += fan_sp_skin + this_fan + "\n"
+                            current_fan_speed = str(fan_sp_skin.split("S")[1])
+                        elif line == ";TYPE:SUPPORT":
+                            modified_data += line + "\n"
+                            modified_data += fan_sp_support + this_fan + "\n"
+                            current_fan_speed = str(fan_sp_support.split("S")[1])
+                        elif ";TYPE:SUPPORT-INTERFACE" in line:
+                            modified_data += line + "\n"
+                            modified_data += fan_sp_support_interface + this_fan + "\n"
+                            current_fan_speed = str(fan_sp_support_interface.split("S")[1])
+                        elif ";MESH:NOMESH" in line or ";MESH:NONMESH" in line:  #compatibility with 5.3.0 'NOMESH'
+                            if fan_combing == True:
+                                modified_data += line + "\n"
+                                modified_data += "M106 S0" + this_fan + "\n"
+                                current_fan_speed = "0"
+                            else:
+                                modified_data += line + "\n"
+                        elif ";TYPE:PRIME-TOWER" in line:
+                            modified_data += line + "\n"
+                            modified_data += fan_sp_prime_tower + this_fan + "\n"
+                            current_fan_speed = str(fan_sp_prime_tower.split("S")[1])
+                        elif line == ";BRIDGE":
+                            modified_data += line + "\n"
+                            modified_data += fan_sp_bridge + this_fan + "\n"
+                            current_fan_speed = str(fan_sp_bridge.split("S")[1])
+                        #If an end layer is defined - Insert the final speed and set the other variables to Final Speed to finish the file
+                        elif line == ";LAYER:" + str(the_end_layer):
```

Comment:
For the similar bits of code here, you can make the three (often) repeated lines into a function, then do the same 'trick' I described earlier with the mapping (and/or another lambda/function).

Reply:
I have no idea what a lambda is beyond it being a Greek letter.